doc/source/data/getting-started.rst

```diff
@@ -1,229 +1,86 @@
-
-    # Create a Dataset of Python objects.
-    ds = ray.data.range(10000)
-    # -> Dataset(num_blocks=200, num_rows=10000, schema=<class 'int'>)
-
-    ds.take(5)
-    # -> [0, 1, 2, 3, 4]
-
-    ds.count()
-    # -> 10000
+Creating and Saving Datasets
+----------------------------
 
-    # Create a Dataset of Arrow records.
-    ds = ray.data.from_items([{"col1": i, "col2": str(i)} for i in range(10000)])
-    # -> Dataset(num_blocks=200, num_rows=10000, schema={col1: int64, col2: string})
+You can create a Dataset from Python objects. These objects can be held inside Dataset as the plain Python objects (where the schema is a Python type), or as Arrow records (in which case their schema is Arrow).
 
-    ds.show(5)
-    # -> {'col1': 0, 'col2': '0'}
-    # -> {'col1': 1, 'col2': '1'}
-    # -> {'col1': 2, 'col2': '2'}
-    # -> {'col1': 3, 'col2': '3'}
-    # -> {'col1': 4, 'col2': '4'}
+.. literalinclude:: ./doc_code/quick_start.py
+   :language: python
+   :start-after: __create_from_python_begin__`
+   :end-before: __create_from_python_end__
 
-    ds.schema()
-    # -> col1: int64
-    # -> col2: string
-
-Datasets can be created from files on local disk or remote datasources such as S3.
+Datasets can also be created from files on local disk or remote datasources such as S3.
 Any filesystem `supported by pyarrow <http://arrow.apache.org/docs/python/generated/pyarrow.fs.FileSystem.html>`__
 can be used to specify file locations:
 
-.. code-block:: python
-
-    # Read a directory of files in remote storage.
-    ds = ray.data.read_csv("s3://bucket/path")
-
-    # Read multiple local files.
-    ds = ray.data.read_csv(["/path/to/file1", "/path/to/file2"])
-
-    # Read multiple directories.
-    ds = ray.data.read_csv(["s3://bucket/path1", "s3://bucket/path2"])
-
-Finally, you can create a ``Dataset`` from existing data in the Ray object store or Ray-compatible distributed DataFrames:
-
-.. code-block:: python
-
-    import pandas as pd
-    import dask.dataframe as dd
-
-    # Create a Dataset from a list of Pandas DataFrame objects.
-    pdf = pd.DataFrame({"one": [1, 2, 3], "two": ["a", "b", "c"]})
-    ds = ray.data.from_pandas([pdf])
-
-    # Create a Dataset from a Dask-on-Ray DataFrame.
-    dask_df = dd.from_pandas(pdf, npartitions=10)
-    ds = ray.data.from_dask(dask_df)
-
-Saving Datasets
-===============
-
-Datasets can be written to local or remote storage using ``.write_csv()``, ``.write_json()``, and ``.write_parquet()``.
-
-.. code-block:: python
+.. literalinclude:: ./doc_code/quick_start.py
+   :language: python
+   :start-after: __create_from_files_begin__`
+   :end-before: __create_from_files_end__
 
-    # Write to csv files in /tmp/output.
-    ray.data.range(10000).write_csv("/tmp/output")
-    # -> /tmp/output/data0.csv, /tmp/output/data1.csv, ...
+Once you have a Dataset, you can save it to local or remote storage in desired format, using ``.write_csv()``, ``.write_json()``, and ``.write_parquet()``.
```

Comment:
We should link out to the file formats section of the "Saving Datasets" feature guide here.